DocNet.Console/Program.cs

```diff
@@ -1,15 +1,259 @@
+        }
+
+        //Get all .cs files from solution or csproj file
+        /*static public string[] ProjectParser(string projectpath, string filetype)
+        {
+            //TODO TEMP CODE PLEASE IGNORE
+            string[] filelist = new string[1];
+            filelist[0] = "-1";
+            //TODO END TEMP CODE
+            return filelist;
+        } */
+
+        //Get the .CS files contained in the file or directory
+        static public string[] GetCsFiles(string inputFile, string outputFile, bool recurseOption)
+        {
+            string[] filelist;
+            string[] nolist = new string[1];
+            nolist[0] = "-1";
+
+            //Check path
+            if (File.Exists(inputFile))
+            {
+                //TODO TEST CODE
+                // This path is a file
+                Console.WriteLine("FILE");
+                //TODO END TEST CODE
+                //If input file is .cs return only the .cs path
+                if (Path.GetExtension(inputFile).Equals(".cs"))
+                {
+                    nolist[0] = inputFile;
+                    //TODO DocNet.DocumentCSFiles(outputFile, nolist)
+                    return nolist;
+                }
+                //If input file is .sln get .cs file list from solution file
+                else if (Path.GetExtension(inputFile).Equals(".sln"))
+                {
+                    Console.WriteLine("Solution File");
+                 
+                   //TODO DocNet.DocumentSolution(outputFile, inputFile)
+                    //filelist = ProjectParser(inputFile, ".sln");
+                    //return filelist;
+                    //TODO TEMP RETURN
+                    return nolist;
+                }
+                //If input file is .csproj get .cs file list from project file
+                else if (Path.GetExtension(inputFile).Equals(".csproj"))
+                {
+                    Console.WriteLine("csproj file");
+                    //TODO DocNet.DocumentCsProject(outputFile, inputFile)
+
+                    //TODO TEMP RETURN
+                    return nolist;
+                }
+                else
+                {
+                    Console.WriteLine("Input file type is invalid. Please input a directory, .cs file, .sln file or .csproj file");
+                    return nolist;
+                }
+
+            }
+            else if (Directory.Exists(inputFile))
+            {
+                //TODO TEST CODE
+                // This path is a directory
+                Console.WriteLine("Directory");
+                //TODO END TEST CODE
+                if (recurseOption)
+                {
+                    //Recursively search directory for .cs files and return a list
+                    filelist = Directory.GetFiles(inputFile, "*.cs", SearchOption.AllDirectories);
+                    if (filelist.Length > 0)
+                    {
+                        //TODO DocNet.DocumentCSFiles(outputFile, filelist)
```

Comment:
Isn't all the recurision contained in " filelist = Directory.GetFiles(inputFile, "*.cs", SearchOption.AllDirectories); "? So it would be called after the recursion finished?